fblock.go

```diff
@@ -0,0 +1,455 @@
+// MIT License
+//
+// Copyright 2018 Canonical Ledgers, LLC
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+
+package factom
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"fmt"
+	"sort"
+
+	merkle "github.com/AdamSLevy/go-merkle"
+	"github.com/Factom-Asset-Tokens/factom/varintf"
+)
+
+const (
+	FBlockMinuteMarker = 0x00
+)
+
+// FBlock represents a Factom Factoid Block
+type FBlock struct {
+	KeyMR       *Bytes32 `json:"keymr"`
+	LedgerKeyMR *Bytes32 `json:"ledgerkeymr"`
+
+	Header FBlockHeader `json:"header"`
+
+	Transactions []*FactoidTransaction `json:"transactions"`
```

Comment:
What is your argument for not using them?

I noticed you had not (noticed today when I am actually using the repo). It makes things like this work. 
```
for _, tx := range fblock.Transactions {
     tx.Get(c)
}
```
Without you have to do:
```
for i, _ := range fblock.Transactions {
     fblock.Transactions[i].Get(c)
}
```

I just don't see the benefit of not using a pointer in the slice, except for the potential bug.

Reply:
yes and that is how it should be. My coding mantra, stolen from Bill Kenedy, is "don't make things easy to do, make things easy to understand". Using a pointer hides the fact that you are modifying the underlying slice in that value semantic for loop. 

Seeing this:  (don't use that second underscore in this case btw)
```
for i := range fblock.Transactions {
     fblock.Transactions[i].Get(c)
}
```
makes it very clear that my intent is to index into the slice I'm ranging over. If I need to reuse that same tx a lot I do this at the top:
```
for i := range fblock.Transactions {
     tx := &fblock.Transactions[i]
    tx.Get(c)
 ...
}
```

Seeing this: 
```
for _, tx := range fblock.Transactions {
...
}
```
makes it really clear that I'm only interested in the values within the slice, and not in modifying the slice itself.

These are features of golang I want to be properly leveraged for clarity. Using a slice of pointers obfuscates that clarity that the syntax is allowing us to take advantage. So it will deceive people unless they go back and read the slice definition, which they might have skipped over because it is not obviously that there is anything extra to pay attention to.

Reply:
I can live with it. When I was first using this library, I fell for that pitfall, and that is when I realized you didn't use pointers for the eblocks -> entries. I probably would have copied that if I had noticed that before I made the transactions here.